src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNull.java

```diff
@@ -74,83 +80,95 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
-                        if (firstArgument instanceof J.Identifier) {
-                            JavaType.Variable fieldType = ((J.Identifier) firstArgument).getFieldType();
-                            return fieldType != null && fieldType.hasFlags(Flag.Static, Flag.Final);
+                    private boolean isCompatibleArgument(J.MethodInvocation m) {
+                        if (!m.getArguments().isEmpty()) {
+                            final Expression firstArgument = m.getArguments().get(0);
+                            if (firstArgument instanceof J.Literal) {
+                                return true;
+                            } else if (firstArgument instanceof J.FieldAccess) {
+                                return isStaticOrFinal(((J.FieldAccess) firstArgument).getName().getFieldType());
+                            } else if (firstArgument instanceof J.Identifier) {
+                                return isStaticOrFinal(((J.Identifier) firstArgument).getFieldType());
+                            }
                         }
                         return false;
                     }
 
+                    private boolean isStaticOrFinal(JavaType.@Nullable Variable fieldType) {
+                        return fieldType != null && fieldType.hasFlags(Flag.Static, Flag.Final);
+                    }
+
                     private boolean isStringComparisonMethod(J.MethodInvocation methodInvocation) {
                         return EQUALS_STRING.matches(methodInvocation) ||
-                                (EQUALS_OBJECT.matches(methodInvocation) && TypeUtils.isString(methodInvocation.getArguments().get(0).getType()))||
+                                (EQUALS_OBJECT.matches(methodInvocation) &&
+                                        TypeUtils.isString(methodInvocation.getArguments().get(0).getType())) ||
                                 EQUALS_IGNORE_CASE.matches(methodInvocation) ||
                                 CONTENT_EQUALS.matches(methodInvocation);
                     }
 
                     private void maybeHandleParentBinary(J.MethodInvocation m, final Tree parent) {
-                        if (parent instanceof J.Binary) {
-                            if (((J.Binary) parent).getOperator() == J.Binary.Type.And &&
-                                    ((J.Binary) parent).getLeft() instanceof J.Binary) {
-                                J.Binary potentialNullCheck = (J.Binary) ((J.Binary) parent).getLeft();
-                                if (isNullLiteral(potentialNullCheck.getLeft()) &&
-                                        matchesSelect(potentialNullCheck.getRight(), requireNonNull(m.getSelect())) ||
-                                    isNullLiteral(potentialNullCheck.getRight()) &&
-                                            matchesSelect(potentialNullCheck.getLeft(), requireNonNull(m.getSelect()))) {
-                                    doAfterVisit(new JavaVisitor<ExecutionContext>() {
```

Comment:
3 ifs (indentations are 3 concerns) and countless conditions later in **one block/context** we doing some `doAfterVisit` is just too much.